modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -410,55 +416,78 @@ CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
     {
       if (assembly != _empty_key)
       {
-        std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>> pin_region_id_map =
-            getMeshProperty<
-                std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>>(
-                RGMB::pin_region_id_map, assembly);
-        for (auto pin = pin_region_id_map.begin(); pin != pin_region_id_map.end(); ++pin)
-          if (_pin_region_id_map.find(pin->first) == _pin_region_id_map.end())
-            _pin_region_id_map.insert(
-                std::pair<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>(
-                    pin->first, pin->second));
-
-        std::map<subdomain_id_type, std::vector<std::vector<std::string>>> pin_block_name_map =
-            getMeshProperty<std::map<subdomain_id_type, std::vector<std::vector<std::string>>>>(
-                RGMB::pin_block_name_map, assembly);
-        for (auto pin = pin_block_name_map.begin(); pin != pin_block_name_map.end(); ++pin)
-          if (_pin_block_name_map.find(pin->first) == _pin_block_name_map.end())
-            _pin_block_name_map.insert(
-                std::pair<subdomain_id_type, std::vector<std::vector<std::string>>>(pin->first,
-                                                                                    pin->second));
-
-        // Define background and duct region ID map from constituent assemblies
         subdomain_id_type assembly_type =
             getMeshProperty<subdomain_id_type>(RGMB::assembly_type, assembly);
-        if (_background_region_id_map.find(assembly_type) == _background_region_id_map.end())
+        if (!getMeshProperty<bool>(RGMB::is_control_drum, assembly))
+        {
+          // For assembly structures, store region ID and block names of assembly regions and
+          // constituent pins
+          std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>
+              pin_region_id_map = getMeshProperty<
+                  std::map<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>>(
+                  RGMB::pin_region_id_map, assembly);
+          for (auto pin = pin_region_id_map.begin(); pin != pin_region_id_map.end(); ++pin)
+            if (_pin_region_id_map.find(pin->first) == _pin_region_id_map.end())
```

Comment:
Have you tried using range `insert` directly. I think it checks uniqueness anyway.

Reply:
Removed the if statement